examples/dbapi_proxy_streaming.py

```diff
@@ -0,0 +1,43 @@
+import logging
+import asyncio
+import sqlite3
+from mysql_mimic import (
+    MysqlServer,
+    Session,
+)
+
+logger = logging.getLogger(__name__)
+
+
+class DbapiProxyStreamingSession(Session):
+    def __init__(self):
+        super().__init__()
+        self.conn = sqlite3.connect(":memory:")
+
+    async def query(self, expression, sql, attrs):
+        cursor = self.conn.cursor()
+        cursor.execute(expression.sql(dialect="sqlite"))
+        if cursor.description:
+            column_names = [column[0] for column in cursor.description]
+            return self.fetch_data(cursor), column_names
+        return None
+
+    async def fetch_data(self, cursor):
+        while True:
+            rows = cursor.fetchmany(1000)
+            for r in rows:
+                yield r
+
+            # not sure if this is the right place for closing the cursor
+            cursor.close()
+            break
```

Comment:
This method only ever returns 1000 rows, since it `breaks` after the first iteration.

I think you want something like:
```python
while (rows := cursor.fetchmany(1000)):
    for r in rows:
        yield r
```

Reply:
Thank you so much for your feedback, and thanks for your patience!

I will rework the example with your suggestion.